src/pages/steps/modals/PriceInfoModal.tsx

```diff
@@ -0,0 +1,312 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect, useRef, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+
+import i18n from '@/i18n/index';
+import type { Values } from '@/types/Values';
+import { Alert, AlertVariants } from '@/ui/Alert';
+import { Box, parseSpacing } from '@/ui/Box';
+import { Button, ButtonVariants } from '@/ui/Button';
+import { FormElement } from '@/ui/FormElement';
+import { Icons } from '@/ui/Icon';
+import { Inline } from '@/ui/Inline';
+import { Input } from '@/ui/Input';
+import { Modal, ModalSizes, ModalVariants } from '@/ui/Modal';
+import { Stack } from '@/ui/Stack';
+import { Text, TextVariants } from '@/ui/Text';
+
+const PRICE_CURRENCY = 'EUR';
+
+const PriceCategories = {
+  BASE: 'base',
+  TARIFF: 'tariff',
+} as const;
+
+type PriceCategory = Values<typeof PriceCategories>;
+
+type NameInLanguages = Partial<{
+  nl: string;
+  fr: string;
+  de: string;
+  en: string;
+}>;
+
+type Rate = {
+  name: NameInLanguages;
+  category: PriceCategory;
+  price: string;
+  priceCurrency: string;
+};
+
+type FormData = { rates: Rate[] };
+
+type PriceInfoModalProps = {
+  visible: boolean;
+  onClose: () => void;
+  onSubmitValid: (data: FormData) => Promise<void>;
+  priceInfo: Rate[];
+};
+
+const isNotUitpas = (value: any): boolean => {
+  return value[i18n.language].toLowerCase() !== 'uitpas';
+};
+
+const shouldHaveAName = (value: any): boolean => {
+  return !!value[i18n.language];
+};
+
+const priceRegex: RegExp = /^([1-9][0-9]*|[0-9]|[0])(,[0-9]{1,2})?$/;
```

Comment:
```suggestion
const PRICE_REGEX: RegExp = /^([1-9][0-9]*|[0-9]|[0])(,[0-9]{1,2})?$/;
```

This can be a constant at the top of the file